src/main/java/racing/domain/RacingGame.java

```diff
@@ -1,81 +1,46 @@
+    private static final String COMMA = ", ";
 
-    private final List<Car> cars;
+    private final MovingStrategy movingStrategy;
+    private final List<List<Car>> raceStates = new ArrayList<>();
+    private final Cars cars;
 
-    public RacingGame() {
-        this.cars = new ArrayList<>();
+    public RacingGame(Cars cars) {
+        movingStrategy = new RandomMovingStrategy();
+        this.cars = cars;
     }
 
-    public void startGame() {
-        try {
-            final List<String> nameList = UserInput.getCarNames();
-            for (String name : nameList) {
-                cars.add(new Car(name));
-            }
-            checkCarCount(cars);
-
-            final int tryNumber = UserInput.getTryNumber();
-            startRacing(tryNumber);
-            showGameResult();
-        } catch (IllegalArgumentException e) {
-            System.out.println(e.getMessage());
-            startGame();
-        }
-    }
-
-    private static void checkCarCount(List<Car> cars) {
-        if (cars.isEmpty()) {
-            throw new IllegalArgumentException("[ERROR] 최소 1대 이상의 자동차가 필요합니다.");
-        }
-    }
-
-    private void startRacing(final int tryNumber) {
-        System.out.println("\n실행 결과");
-        for (int i = 0; i < tryNumber; i++) {
-            driveCars();
-            System.out.println();
-        }
+    public static RacingGame registerCars(List<String> names) {
+        return new RacingGame(new Cars(names, 0));
     }
 
-    private void driveCars() {
-        for (Car car : cars) {
-            car.drive(generateRandomNumber());
+    public void race(final TryNumber tryNumber) {
+        while (tryNumber.hasNextStep()) {
+            raceStates.add(cars.driveCars(movingStrategy));
+            tryNumber.nextStep();
         }
     }
 
-    private int generateRandomNumber() {
-        return (int) (Math.random() * RANGE_OF_RANDOM_NUMBER);
+    public List<List<Car>> raceResult() {
+        return new ArrayList(new ArrayList(raceStates));
     }
 
-    private void showGameResult() {
-        int maxPosition = 0;
-        for (Car car : cars) {
-            maxPosition = setMaxPosition(maxPosition, car.getPosition());
-        }
-
-        final List<String> winners = new ArrayList<>();
-        for (Car car : cars) {
-            winners.add(car.getNameBy(maxPosition));
-        }
-        winners.removeAll(Arrays.asList("", null));
-
-        final String resultOfWinners = String.join(", ", winners);
-        System.out.println("최종 우승자 : " + resultOfWinners);
-    }
-
-    private int setMaxPosition(int maxPosition, int carPosition) {
-        if (carPosition > maxPosition) {
-            return carPosition;
-        }
-        return maxPosition;
+    public String decideWinners() {
+        return cars.getCars().stream()
+            .filter(car -> car.isSamePosition(new Position(cars.getMaxPosition())))
+            .map(Car::getNameValue)
+            .collect(Collectors.joining(COMMA));
```

Comment:
매번 cars 객체에게 `maxPosition`을 확인해서 반환하라는 메시지를 보내고 있어요! 또한 매번 Position 객체를 생성하고 있어요. 🥲
아래와 같이 변경해보면 어떨까요? 😉

```suggestion
        Position maxPosition = cars.findMaxPosition();
        return cars.getCars().stream()
            .filter(car -> car.isSamePosition(maxPosition))
            .map(Car::getNameValue)
            .collect(Collectors.joining(COMMA));
```

추가로 이 또한 `String` 타입으로 바로 만들지 말고 List<String> 혹은 List<Car> 타입으로 반환한 뒤, Controller에서 OutputView의 메서드 인자로 넘기면 View 단에서 출력을 위해 `,` 붙인다거나 하는 작업을 하도록 변경해주세요! 🙏

Reply:
오.. 정말 매번 Position 객체를 생성하고 있네요! int형 타입을 반환하던 `getMaxPosition`을 말씀해주신대로 Position 타입을 리턴하는 `findMaxPosition`으로 변경하여 코드에 반영하겠습니다! 
또한, ',' 로 조인하는 부분은 출력에서 담당해야하는 것이므로 이것 또한 코멘트 남겨주신대로 수정하도록 하겠습니다!